src/marlin_stubs/pause/M600.cpp

```diff
@@ -116,10 +119,48 @@ void GcodeSuite::M600() {
 }
 
 void M600_execute(xyz_pos_t park_point, int8_t target_extruder,
-    xyze_float_t resume_point, std::optional<float> unloadLength, std::optional<float> fastLoadLength, std::optional<float> retractLength,
-    pause::Settings::CalledFrom);
+    xyze_float_t resume_point, std::optional<float> unloadLength, std::optional<float> fastLoadLength,
+    std::optional<float> retractLength, std::optional<filament::Colour> filament_colour,
+    std::optional<filament::Type> filament_type, pause::Settings::CalledFrom);
 
 void M600_manual() {
+    char filamenttype[16] = { '\0' };
+    char colourtype[16] = { '\0' };
+
+    auto filament_to_be_loaded = filament::Type::NONE;
+    const char *text_begin = 0;
+    if (parser.seen('S')) {
+        text_begin = strchr(parser.string_arg, '"');
+        if (text_begin) {
+            ++text_begin; // move pointer from '"' to first letter
+            const char *text_end = strchr(text_begin, '"');
+            if (text_end) {
+                auto filament = filament::get_type(text_begin, text_end - text_begin);
+                if (filament != filament::Type::NONE) {
+                    filament_to_be_loaded = filament;
+                }
+            }
+        }
+    }
+
+    if (parser.seenval('C')) {
+        const char *colourtype_ptr = nullptr;
+        if ((colourtype_ptr = strstr(parser.string_arg, " C")) != nullptr) {
+            if (*(colourtype_ptr + 2) == '"' || *(colourtype_ptr + 2) == ' ') {
```

Comment:
I'm a bit nervous about these pointer arithmetics with `+2` (here and 3 lines below). What guarantees this is in-range? If there's a guarantee from something, a comment here explaining it would be nice.

Reply:
Right, I should add a range check for the whole string length.

Reply:
The +2 referred to " C" which is indeed always 2, but might lead to other issues. So I decided to restructure the whole part. Solved with commit [d631509](https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/3581/commits/d6315094ec575102dc032859100f6114a28a341d)